HW02_A_Graph-Fails.Rmd

```diff
@@ -1,87 +1,146 @@
+
+#### My corrected graph 2:
+```{r}
+ggplot(data = mpg) + 
+  geom_point(mapping = aes(x = displ, y = hwy), color = "blue")
+```
+<br> Either code works
+```{r}
+ggplot(mpg, aes(x=displ, y=hwy)) + 
+  geom_point(color="blue")
+```
+
+### Graph Fail 3
+Two mistakes in this graph. First, I wanted to make the the points slightly bolder, but changing the alpha to 2 does nothing. What does alpha do and what does setting it to 2 do? What could be done instead if I want the points slightly bigger? 
+
+<br>**My answer: Alpha (has values between 0-1, where 0=invisible and 1=darkest) that controls transparency of points in 'geom_point', for example. Setting alpha to 2 would not do anything more than setting it to 1. To make the points slightly bigger, the 'size' aesthetic can bet set as an argument within 'geom_point'.**
+
+Second, I wanted to move the legend on top of the graph since there aren't any points there, putting it at approximately the point/ordered pair (5, 40). How do you actually do this? Also, how do you remove the legend title ("class")? Finally, how would you remove the plot legend completely?
+
+<br>**My answer: To change stylistic elements of a plot, call 'theme()' and change 'legend.position' using 'c(x, y)', where 'c(0, 0)' means bottom-left of the coordinate plane and 'c(0,1)' means top-right. To remove the legend title ("class"), use the 'element_blank()' function in a theme layer: theme(legend.title=element_blank()). To remove the plot legend completely: theme(legend.position="none"). Alternatively, would theme(legend.position=element_blank()) also work??**
+```{r}
+mpg %>% 
+ggplot() + 
+  geom_point(mapping = aes(x = displ, y = hwy, color = class), alpha = 2) + 
+  theme(legend.direction = "horizontal") + 
+  theme(legend.position = c(5, 40))
+
+```
+
+#### My corrected graph 3:
+```{r}
+mpg %>% 
+ggplot() + 
+  geom_point(mapping = aes(x = displ, y = hwy, color = class), size = 3) + 
+  theme(legend.direction = "horizontal") + 
+  theme(legend.position = c(.6, .8)) +
+  theme(legend.title = element_blank())
+
+```
+
+### Graph Fail 4
+I wanted just one smoothing line. Just one line, to show the general relationship here. But that's not happening. Instead I'm getting 3 lines, why and fix it please?
+
+<br>**My answer: Since 'color=drv' is defined within the ggplot function, it is applied to all layers of the graph, including the smoothing lines. To tell ggplot you don't want the lines to be affected by the different categories of 'drv', it should only be defined within 'geom_point'**
+
+```{r}
+mpg %>% 
+ggplot(mapping = aes(x = displ, y = hwy, color = drv)) + 
+  geom_point() + 
+  geom_smooth(se = F,) #se = F makes it so it won't show the error in the line of fit
+```
+
+#### My corrected graph 4:
+```{r}
+mpg %>% 
+ggplot(mapping = aes(x = displ, y = hwy)) + 
+  geom_point(mapping=aes(color = drv)) + 
+  geom_smooth(se = F, color="yellow") #se = F makes it so it won't show the error in the line of fit
```

Comment:
Hi Monty,

I ran your code, and it is giving the same result as mine! what I have for the last line: geom_smooth(se = F, aes(color = NULL))
so it just commended to ignore the color differences from groups, and draw the line.

Your way def works well too!